src/mindlessgen/qm/tm.py

```diff
@@ -0,0 +1,223 @@
+                temp_path=temp_path, arguments=arguments
+            )
+            if verbosity > 2:
+                print(tm_log_out)
+            if return_code != 0:
+                raise RuntimeError(
+                    f"Turbomole failed with return code {return_code}:\n{tm_log_err}"
+                )
+
+            # # revert the coord file to xyz file
+            xyzfile = Path(temp_path / "coord").resolve()
+            optimized_molecule = molecule.copy()
+            optimized_molecule.read_xyz_from_coord(xyzfile)
+            return optimized_molecule
+
+    def singlepoint(self, molecule: Molecule, verbosity: int = 1) -> str:
+        """
+        Perform a single point calculation using Turbomole.
+        """
+        # Create a unique temporary directory using TemporaryDirectory context manager
+        with TemporaryDirectory(prefix="turbomole_") as temp_dir:
+            temp_path = Path(temp_dir).resolve()
+            # write the molecule to a temporary file
+            molfile = "coord"
+            molecule.write_coord_to_file(temp_path / molfile)
+            print(temp_path / molfile)
+
+            # write the input file
+            inputname = "control"
+            tm_input = self._gen_input(molecule)
+            if verbosity > 1:
+                print("Turbomole input file:\n##################")
+                print(self._gen_input(molecule))
+                print("##################")
+            with open(temp_path / inputname, "w", encoding="utf8") as f:
+                f.write(tm_input)
+
+            # set up the turbomole single point calculation command
+            run_tm = ["PARNODES=1 ridft > ridft.out"]
+
+            tm_log_out, tm_log_err, return_code = self._run(
+                temp_path=temp_path, arguments=run_tm
+            )
+            if verbosity > 2:
+                print(tm_log_out)
+            if return_code != 0:
+                raise RuntimeError(
+                    f"Turbomole failed with return code {return_code}:\n{tm_log_err}"
+                )
+
+            return tm_log_out
+
+    def check_gap(
+        self, molecule: Molecule, threshold: float, verbosity: int = 1
+    ) -> bool:
+        """
+        Check if the HL gap is larger than a given threshold.
+        """
+        raise NotImplementedError("check_gap not implemented for turbomole.")
+
+    def _run(self, temp_path: Path, arguments: list[str]) -> tuple[str, str, int]:
+        """
+        Run turbomole with the given arguments.
+
+        Arguments:
+        arguments (list[str]): The arguments to pass to turbomole.
+
+        Returns:
+        tuple[str, str, int]: The output of the turbomole calculation (stdout and stderr)
+                              and the return code
+        """
+        try:
+            tm_out = sp.run(
+                arguments,
+                cwd=temp_path,
+                capture_output=True,
+                check=True,
+                shell=True,
```

Comment:
```suggestion
                shell=True, # shell=True is necessary for inserting the `PARNODES=xx` command, unfortunately.
```
Add this, if you agree.